code/game/objects/structures/morgue.dm

```diff
@@ -268,13 +268,19 @@ GLOBAL_LIST_EMPTY(crematoriums)
 			else
 				M.log_message("was cremated", LOG_ATTACK)
 
+			for(var/obj/item/I in M)
+				if(I.resistance_flags & INDESTRUCTIBLE)
+					M.dropItemToGround(I, TRUE)
+
```

Comment:
```suggestion
```

`GetAllContents()` is called at building `conts`, so obj loop below can handle contents of any mob.

Reply:
`qdel` running through `content` was found buried deep in the second `/atom/movable/Destroy` (yes there really are two of them) which leaves below iteration over `conts` in a weird place (why does the code call qdel on every item in the first place when most of the `conts` are covered by `qdel`ing mobs?)

So my suggestion is:

1. Iterate over `conts` to move `INDESTRUCTIBLE` items out of mob/obj before any `qdel`
2. Then iterate over `contents` for objs (i.e. objects that won't be covered by mob `qdel`) after mob `qdel` to sort out `INDESTRUCTIBLE` items for the second time

Reply:
Alright